timemachine/fe/atom_mapping.py

```diff
@@ -236,6 +243,78 @@ def _deduplicate_all_cores(all_cores):
     return list(unique_cores.values())
 
 
+def ring_forming_breaking_count(mol_a, mol_b, core: NDArray) -> int:
+    """Counts the number of rings broken and formed given a core. Does not count ring insertion/deletions. Considers
+    the number number of rings broken/formed in both endstates, reversing the mol order and the core order produces
+    the same value.
+
+    The algorithm is to add all ring atoms/bonds to a graph, and count the cycles that contain both core and mol a or
+    mol b atoms.
+    """
+    g = nx.Graph()
+    core_a_to_b = {a: b for a, b in core}
+    core_b_to_a = {b: a for a, b in core}
+    for atom in mol_a.GetAtoms():
+        idx = atom.GetIdx()
+        if atom.IsInRing():
+            if idx in core_a_to_b:
+                g.add_node(idx, atom_type=AtomMapFlags.CORE)
+            else:
+                g.add_node(idx, atom_type=AtomMapFlags.MOL_A)
+    for atom in mol_b.GetAtoms():
+        idx = atom.GetIdx()
+        if atom.IsInRing():
+            if idx in core_b_to_a:
+                g.add_node(idx + mol_a.GetNumAtoms(), atom_type=AtomMapFlags.CORE)
```

Comment:
I'm a bit confused here. Are the nodes in the graph intended to be labeled according to the combined mol atom index? If so, should this use `atom_map.b_to_c`?

Reply:
Per offline discussion moved to using alchemical indices: https://github.com/proteneer/timemachine/pull/1363/commits/8728b64ecc6fb30ff546bd1209016e31afd0091d